internal/controller/cluster/rack.go

```diff
@@ -974,6 +975,42 @@ func (r *SingleClusterReconciler) scaleDownRack(
 		); !res.IsSuccess {
 			return found, res
 		}
+
+		// Wait for migration to complete before deleting the pods.
+		if res := r.waitForMigrationToComplete(
+			ignorablePodNames,
+		); !res.IsSuccess {
+			r.Log.Error(
+				res.Err, "Failed to wait for migration to complete before deleting pods",
+				"rackID", rackState.Rack.ID,
+			)
+
+			return found, res
+		}
+	}
```

Comment:
In case of failed pod also, I think we should wait for migration to complete. What if the pods recently failed and migration are still going on. We should make sure the migration are complete in any case